velox/exec/RowNumber.cpp

```diff
@@ -250,7 +225,19 @@ FlatVector<int64_t>& RowNumber::getOrCreateRowNumberVector(vector_size_t size) {
 
 RowVectorPtr RowNumber::getOutput() {
   if (input_ == nullptr) {
-    return nullptr;
+    if (spillInputReader_ == nullptr) {
+      return nullptr;
+    }
+
+    if (yield_) {
+      VELOX_CHECK_NULL(input_);
```

Comment:
nit: do we need this check? since we only reach here if condition on L227 is true where input_ == nullptr